servers/sunrpc/server.go

```diff
@@ -31,7 +42,7 @@ func New(l net.Listener) *SunRPC {
 	programsList = []Program{
 		newGfHandshake(),
 		newGfDump(),
-		newGfPortmap(),
+		pmap.NewGfPortmap(),
```

Comment:
> _ "github.com/gluster/glusterd2/pmap"
> init() func is called when pmap is imported, which adds to programsList

IMHO this is not a good convention to follow in Go for this purpose. It'll work though. Very often, being explicit is more helpful to new people reading your code.

Here's why I prefer this format:

```
programsList = []Program{
 		newGfHandshake(),
 		newGfDump(),
		pmap.NewGfPortmap(),}
```

* If all plugins are imported and added at once place in `servers/sunrpc` or `servers/sunrpc/program` package, it's clear to see list of plugins which hook into sunrpc server. I don't have to look at multiple packages. I can also comment out and disable the ones I don't want during my development, all at one place.
* If the plugin resides outside of the repo, those plugins can also be used, simply by importing it them here and explicitly adding the program package
* The constructor (example `NewGfPortmap`) can be passed arguments (like config or env variable). The plugin won't have to share context/state with glusterd2.
* Every plugin won't have to import `github.com/gluster/glusterd2/servers/sunrpc/program`.